core/src/main/java/discord4j/core/shard/GatewayBootstrap.java

```diff
@@ -502,7 +519,20 @@
                     StatusUpdate initial = Optional.ofNullable(initialPresence.apply(shard))
                             .map(Presence::asStatusUpdate)
                             .orElse(null);
-                    IdentifyOptions identify = new IdentifyOptions(shard, initial, guildSubscriptions);
+                    final IdentifyOptions identify;
+                    if (intents == null) {
```

Comment:
Personally, I think we should fail if intents are not specified (or require them in some other way) rather than defaulting to all. As far as I understand, the gateway is going to close immediately anyways if we try to specify an intent we aren't allowed to (like presences). Open to discussion here.

Reply:
Guild Subscribtions are not yet disabled, so you probably want to continue support on that. Thats a decision the D4J Team has to make. I'm also happy to drop guildSubscriptions and force intents, because its basically the same, with a wider support on features.